[29.x] Backports

fanquake opened the pull request:
Backports for `29.x`:

- #32184
- #32187
- #32248
- #32286
- #32312
- #32336
- #32353
- #32356
- #32389
- #32437
- #32454
- #32469
- #32483
- #32553
- https://github.com/bitcoin-core/gui/pull/864
- https://github.com/bitcoin-core/gui/pull/868

hebasto committed:
ci: Add workaround for vcpkg's libevent package

This change is necessary for Windows GHA images, which provide
CMake >= 4.0.

Github-Pull: #32184
Rebased-From: ef00a28414daed2dd026b458082ed03fe9508074

Commit:
refactor: Remove spurious virtual from final ~CZMQNotificationInterface

Github-Pull: #32187
Rebased-From: fa69c42fdf0aeec0546e951bc6132ab630edb9d4

furszy committed:
gui: crash fix, disconnect numBlocksChanged() signal during shutdown

The crash stems from the order of the shutdown procedure:
We first unset the client model, then destroy the wallet controller—but we leave
the internal wallet models ('m_wallets') untouched for a brief period. As a result,
there’s a point in time where views still have connected signals and access to
wallet models that are not connected to any wallet controller.
Now.. since the clientModel is only replaced with nullptr locally and not destroyed
yet, signals like numBlocksChanged can still emit. Thus, when wallet views receive
them, they see a non-null wallet model ptr, and proceed to call backend functions
from a model that is being torn down.

As the shutdown procedure begins by unsetting clientModel from all views. It’s safe
to ignore events when clientModel is nullptr.

Github-Pull: gui#864
Rebased-From: 71656bdfaa6bfe08ce9651246a3ef606f923351b

laanwj committed:
Remove support for RNDR/RNDRRS for aarch64 on Linux

This hardware feature is

- rarely supported on SoCs (and broken on like half of the chips that support it in the first place) (#31817)
- apparently not compiled into the release binary (https://github.com/bitcoin/bitcoin/issues/31817#issuecomment-2795885962)
- hard to test in CI, due to unavailable of hardware

Better to remove it.

This reverts commit aee5404e02e203a256c1a97b629b9b107cc8bb07.

Closes #31817.

Github-Pull: #32248
Rebased-From: 7749d929a0d9dfe71541a22e557ea41e01df28ce

BrandonOdiwuor committed:
test: Handle empty string returned by CLI as None in RPC tests

Github-Pull: #32286
Rebased-From: a4041c77f0e20d004524868e70ff12508832c9eb

fanquake committed:
doc: update release notes for 29.x

hebasto commented on .github/workflows/ci.yml:
```diff
@@ -198,6 +198,8 @@ jobs:
           Set-Location "$env:VCPKG_INSTALLATION_ROOT"
           Add-Content -Path "triplets\x64-windows.cmake" -Value "set(VCPKG_BUILD_TYPE release)"
           Add-Content -Path "triplets\x64-windows-static.cmake" -Value "set(VCPKG_BUILD_TYPE release)"
+          # Workaround for libevent, which requires CMake 3.1 but is incompatible with CMake >= 4.0.
+          sed -i '1s/^/set(ENV{CMAKE_POLICY_VERSION_MINIMUM} 3.5)\n/' "${VCPKG_INSTALLATION_ROOT}/scripts/ports.cmake"
```
`bash` has to be used to run `sed`:
```diff
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -66,5 +66,6 @@ jobs:
           Write-Host "PowerShell version $($PSVersionTable.PSVersion.ToString())"
 
       - name: Using vcpkg with MSBuild
+        shell: bash
         run: |
           echo "VCPKG_INSTALLATION_ROOT is '$VCPKG_INSTALLATION_ROOT'"
```

See: f8619196ceb5c6a58125506d276d9515837f043a

enirox001 committed:
test: Fix nTimes typo in feature_pruning test

Fix incorrect variable name in comment (nTimes -> nTime) in
feature_pruning.py. This typo caused the test to always reset
mine_large_blocks.nTime to 0, rather than only on the first run
as intended.

Github-Pull: #32312
Rebased-From: 772ba7f9ce09e836a51636524a8a96a23946d658

enirox001 committed:
test: Use uninvolved pruned node in feature_pruning undo test

After fixing the nTime variable name, the test_pruneheight_undo_presence
test began failing because node 2, which is involved in reorg testing,
could be on a different chain than other nodes. This caused failures
when trying to fetch blocks from other nodes that didn't recognize
node 2's chain.

Switch to using node 5 instead, which is also a pruned node but isn't
involved in reorg testing, ensuring it stays on the same chain as the
other nodes. This allows the block fetching to work as intended in the
test.

Github-Pull: #32312
Rebased-From: 2aa63d511affdcc9980b58fc4ff18b8ad10b0f8c

Commit:
test: Add imports for util bpf_cflags

This is required for the next commit.

Github-Pull: #32336
Rebased-From: fa0c1baaf89805182de56d6a2e8eed2265a94b40

Commit:
scripted-diff: Use bpf_cflags

-BEGIN VERIFY SCRIPT-

 ren() { sed --regexp-extended -i "s/$1/$2/g" $( git grep --extended-regexp -l "$1" ) ; }

 ren 'cflags=\["-Wno-error=implicit-function-declaration"\]' 'cflags=bpf_cflags()'

-END VERIFY SCRIPT-

Github-Pull: #32336
Rebased-From: facb9b327b9da39ce1e09ed56199be9efb19b5b8

laanwj committed:
qt: Replace stray tfm::format to cerr with qWarning

GUI warnings should go to the log, not to the console (which may not be
connected at all).

Github-Pull: gui#868
Rebased-From: edd46566bd66cea7d7f4116429fe1c11d2187ba2

monlovesmango committed:
doc: Fix fuzz test_runner.py path

This commit fixes the path listed in the documentation for the fuzz
testing test_runner.py. Previously the --help option worked but running
fuzz tests from the documented path did not.

Github-Pull: #32353
Rebased-From: 61f238e84ac6d24d8f420c2eabcbb2980d7fcb1e

hebasto commented:
> Could also include [bitcoin-core/gui#864](https://github.com/bitcoin-core/gui/pull/864), possibly?

+1

hebasto committed:
cmake: Respect user-provided configuration-specific flags

Github-Pull: #32356
Rebased-From: edde96376a2961dec3730331b3d171ddf972589f

hebasto commented:
>    * [Replace stray tfm::format to cerr with qWarning bitcoin-core/gui#868](https://github.com/bitcoin-core/gui/pull/868)

It seems `#include <QDebug>` is missed.

monlovesmango committed:
doc: Fix test_bitcoin path

This commit fixes a couple command paths for interacting with the
test_bitcoin binary within the Unit Test documentation.

Github-Pull: #32389
Rebased-From: 6cbc28b8dd629062950f195facc009fd8ba86310

fanquake committed:
crypto: disable ASan for sha256_sse4 with Clang

This can alsofail to compile when optimisations are being used, see:
https://github.com/bitcoin/bitcoin/issues/31913.
So disable just ASan for this function under any optimisation level.

Github-Pull: #32437
Rebased-From: 4e8ab5e00fa72016a7ec0e0505ca025d4e59e4d8

jb55 committed:
tracing: fix invalid argument in mempool_monitor

The mempool_monitor tracing tool is incorrectly reading the reason
as the first argument. Fix this!

Github-Pull: #32454
Rebased-From: 31c5ebc4007884b655f2f90ca09e36e0b9ada4da

hebasto committed:
cmake: Allow `WITH_DBUS` on all Unix-like systems

This change makes the `WITH_DBUS` option available on all Unix-like
systems, not just Linux, thereby fixing a regression that was
overlooked during the migration from Autotools.

Note: Enabling D-Bus support on macOS still makes no sense, since the
`Notificator` class uses the User Notification Center regardless.

Github-Pull: #32469
Rebased-From: 5b7ed460c7c181f1fd34a927a09aed36284083cb

mzumsande committed:
test: Fix intermittent failure in wallet_basic.py

There could be a race with outstanding TxAddedToMempool notifications
being applied to the soon-to-be created wallet.

Fixes an intermittent timeout reproducable by adding a sleep to
AddToWallet.

Github-Pull: #32483
Rebased-From: 07350e204dedfba20da461d9cdcd469dc95e01c3

mzumsande committed:
test: fix another intermittent failure in wallet_basic.py

During init, the test framework will start using rpc after the
mempool was loaded. It will not wait for postInitProcess or
outstanding transactionAddedToMempool notifications, leading to
a possible race, in which listunspent is being called while the
tx is still in Inactive status. Prevent this by processing
outstanding notifications.

Github-Pull: #32483
Rebased-From: e7ad86e1ca3b0b2f2795e91c2f9959486c67dd90

achow101 committed:
walletdb: Log the wallet version after it has been read from disk

Logging the wallet version before anything has been read from disk results
in the wrong version being logged.

Also split the last client version logging as it may not always be
present to be logged.

Github-Pull: #32553
Rebased-From: 359ecd3704993422eb53e3da2a7d0bea2f575ab0

achow101 committed:
test: Check that the correct versions are logged on wallet load

Github-Pull: #32553
Rebased-From: 39a483c8e9dcfe8ec243fa72269e1df9e75059ab

achow101 committed:
test: check that creating a wallet does not log version info

Github-Pull: #32553
Rebased-From: 4b2cd0b41ff4800c8801f2c44883eaec60a035fa

willcl-ark reviewed (approved):
crACK a0d1f69b555fa0c76df1a63a0b127c7816596107

Checked all patches were backported cleanly and appear in the release notes.

Did not build/test.
